### PR TITLE
Mimes french kiss now activates the SM

### PR DESCRIPTION
**Pull request by BumbertoEko**
## About The Pull Request
Before if you blew a kiss at the sm as a mime it did nothing and crashed into it
now it activates the sm like a normal kiss
## Why It's Good For The Game
if your a mime setting up a sm and there is no power
you can activate it using your kiss
## Changelog
:cl:
fix: Mimes french kisses now activate the sm
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
At this point you should just put a `kiss_power` var on the `kiss` subtype and read from there

**Comment by BumbertoEko** [comment]
niko how did u find this stop spying

**Comment by NikolasK5** [comment]
you sended me the link you idiot

**Comment by BumbertoEko** [comment]
nuh uh

**Comment by NikolasK5** [comment]
bumberto caught 4k
nigga your done

**Comment by BumbertoEko** [comment]
dont make me leak the group chat

**Comment by NikolasK5** [comment]
here is a morse code 00011101010100100101110101000

**Comment by NikolasK5** [comment]
NOOOOOOOOOO

**Comment by BumbertoEko** [comment]
ok stop using this as the chat

**Comment by Jacquerel** [comment]
don't do that again or you won't be able to comment on PRs any more

**Comment by Jacquerel** [comment]
> At this point you should just put a `kiss_power` var on the `kiss` subtype and read from there

Also do this so I can merge the PR

**Comment by BumbertoEko** [comment]
i dont know how to do that i never touched byond code but i understand a bit to make this

**Comment by Jacquerel** [comment]
Bascially instead of this:
```
```

You would add a new var called `supermatter_kiss_power` to `/obj/projectile/kiss` and give it an appropriate value (60 by default, 20000 for death kiss, don't need to set one for the mime kiss because it would inherit from the normal kiss)
Then replace that above code with something like:

```
var/kiss_power = 0
if (istype(projectile, /obj/projectile/kiss))
   var/obj/projectile/kiss/smooch = projectile
   kiss_power = smooch.supermatter_kiss_power
```

This means any future kiss would inherit the default value of `60` rather than it _only_ applying to "normal kiss", and can have its value easily adjusted if required

**Review by Ghommie — APPROVED** [review]

File: code/modules/power/supermatter/supermatter_hit_procs.dm [hunk 1]
```diff
@@ -13,11 +13,13 @@
 		return NONE
 
 	var/kiss_power = 0
-	switch(projectile.type)
-		if(/obj/projectile/kiss)
-			kiss_power = 60
-		if(/obj/projectile/kiss/death)
-			kiss_power = 20000
+	var/supermatter_kiss_power = 60
```
Comment (SuperNovaa41) [inline code/modules/power/supermatter/supermatter_hit_procs.dm:16]: i don't understand why you added this

**Pull request merged**